File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int m;
        long int s=0;
        if(x<0)
        {
           while(x!=0)
           {
               m = x%10;
               x = x/10;
               s = s*10+m;
           }
        }
        else
        {
            while(x!=0)
           {
               m = x%10;
               x = x/10;
               s = s*10+m;
           }
        }
        if(s > INT_MAX || s < INT_MIN)
        {
            return 0;
        }
        return s;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        n = x
        sum = 0
        flag = 0
        if(n<0):
            flag = 1
            n = n*(-1)
        while(n>0):
            rem = n%10
            sum = sum*10 + rem
            n = n//10
        
        if(flag==1):
            sum = sum*(-1)

        if(sum>(2**31)-1 or sum<-(2**31)):
            return 0
        return sum